src/main/java/net/mcreator/workspace/misc/WorkspaceInfo.java

```diff
@@ -142,17 +141,18 @@ public Map<String, List<MItemBlock>> getCreativeTabMap() {
 		// ModElement#getGeneratableElement that is not thread safe
 		for (GeneratableElement element : elementsList) {
 			if (element instanceof ITabContainedElement tabElement) {
-				TabEntry tabEntry = tabElement.getCreativeTab();
 				List<MCItem> tabItems = tabElement.getCreativeTabItems();
-				if (tabEntry != null && tabItems != null && !tabItems.isEmpty()) {
-					String tab = tabEntry.getUnmappedValue();
-					if (tab != null && !tab.equals("No creative tab entry")) {
-						if (!tabMap.containsKey(tab)) {
-							tabMap.put(tab, new ArrayList<>());
+				if (tabItems != null && !tabItems.isEmpty()) {
+					for (TabEntry tabEntry : tabElement.getCreativeTabs()) {
+						String tab = tabEntry.getUnmappedValue();
+						if (tab != null) {
+							if (!tabMap.containsKey(tab)) {
+								tabMap.put(tab, new ArrayList<>());
+							}
+
+							tabMap.get(tab).addAll(tabItems.stream().map(e -> new MItemBlock(workspace, e.getName()))

```

Comment:
You could execute this before `for (TabEntry tabEntry : tabElement.getCreativeTabs()) {` as it does not depend on `tabEntry` for some performance gain

Reply:
But this uses data from the loop, should I refactor the loop itself somehow too?

Reply:
`tabMap.get(tabEntry.getUnmappedValue()).addAll(<...>)`

Reply:
`tabItems.stream().map(e -> new MItemBlock(workspace, e.getName())`

this part looks pretty fixed to me